dds/DCPS/transport/rtps_udp/BundlingCacheKey.h

```diff
@@ -0,0 +1,101 @@
+/*
+ *
+ *
+ * Distributed under the OpenDDS License.
+ * See: http://www.opendds.org/license.html
+ */
+
+#ifndef OPENDDS_DCPS_TRANSPORT_RTPS_UDP_BUNDLINGCACHEKEY_H
+#define OPENDDS_DCPS_TRANSPORT_RTPS_UDP_BUNDLINGCACHEKEY_H
+
+#include "Rtps_Udp_Export.h"
+
+#include "dds/DCPS/GuidConverter.h"
+#include "dds/DCPS/PoolAllocator.h"
+#include "dds/DCPS/AddressCache.h"
+#include "dds/DCPS/Hash.h"
+
+#if defined ACE_HAS_CPP11
+#include <functional>
+#endif
+
+OPENDDS_BEGIN_VERSIONED_NAMESPACE_DECL
+
+namespace OpenDDS {
+namespace DCPS {
+
+#pragma pack(push, 1)
+
+struct OpenDDS_Rtps_Udp_Export BundlingCacheKey {
+  BundlingCacheKey(const GUID_t& dst_guid, const GUID_t& from_guid, const GuidSet& to_guids)
+    : dst_guid_(dst_guid)
+    , from_guid_(from_guid)
+    , to_guids_(to_guids)
+  {
+  }
+
+  bool operator<(const BundlingCacheKey& rhs) const
+  {
+    int r = std::memcmp(&dst_guid_, &rhs.dst_guid_, 2 * sizeof (GUID_t));
```

Comment:
This appears to be technically undefined behavior (but we're in uncharted waters with `#pragma pack`) and Coverity will likely complain.  At least change it to use the same kind of `memcmp` that `operator==` has, based on `this`.